net/mwan3/files/usr/sbin/mwan3track

```diff
@@ -41,10 +41,15 @@ if_up() {
 }
 
 validate_track_method() {
+	local ping_version_string
 	case "$1" in
 		ping)
-			if [ -x "/usr/bin/ping" ] && [ "$(/usr/bin/ping -V | grep -o '[0-9]*$')" -gt 20150519 ]; then
+			if [ -x "/usr/bin/ping" ]; then
+				ping_version_string=$(/usr/bin/ping -V)
+			fi
+			if [ -n "$ping_version_string" ] && { [ -z "${ping_version_string##*iputils v*}" ] || [ "$(echo ${ping_version_string} | sed 's/.*iputils s\([0-9]\+\).*/\1/')" -gt 20150519 ] ; }; then
 				# -4 option added in iputils c3e68ac6
+				# change in build between s20200821 and s20190709 causes the binary in OpenWRT package to report a version tag rather than date tag
```

Comment:
why do this ping version check?
we may check it by using it
do ping test before we use it
```
ping -c1 -q -6 ::1 || somewhat
ping -c1 -q -4 127.0.0.1 || somewhat
```
could we use this method the determinate `what ping` ? before the loop.

Reply:
Yes, that would be an alternative solution. I don't see a strong reason to use one way or the other. This was the smaller change that kept with the existing version check logic and works around the iputils package issue. Why do you prefer the test ping method over the version check?

Reply:
emm
The version has countless possibilities including future possibilities, so that the check needs to be updated again
The ping test is more direct and will not cause problems due to subsequent version updates

Reply:
Sure. iputils pings should stably increment its version number, but as the latest iputils-ping.ipk shows, that isn't always translated into the OpenWRT. 

I just added it. It is a bit faster than the `sed` based version I had, but a bit slower than a purely `sh` based implementation of processing the version string. It only happens once on startup and is very fast regardless of which way it is done, so performance isn't a big issue here.